tests/unit-core/test-promise-callback.c

```diff
@@ -0,0 +1,283 @@
+  CP = UINT8_MAX - 1, /**< same as JERRY_PROMISE_CALLBACK_CREATE with Promise value */
+  E = UINT8_MAX, /**< marks the end of the event list */
+} jerry_promise_callback_event_abbreviations_t;
+
+static int user;
+static const uint8_t *next_event_p;
+
+static void
+promise_callback (jerry_promise_callback_notification_types_t notification_type, /**< notification type */
+                  const jerry_value_t object, /**< target object */
+                  const jerry_value_t value, /**< optional argument */
+                  void *user_p) /**< user pointer passed to the callback */
+{
+  TEST_ASSERT (user_p == (void *) &user);
+
+  switch (notification_type)
+  {
+    case JERRY_PROMISE_CALLBACK_CREATE:
+    {
+      TEST_ASSERT (jerry_value_is_promise (object));
+      if (jerry_value_is_undefined (value))
+      {
+        break;
+      }
+
+      TEST_ASSERT (jerry_value_is_promise (value));
+      TEST_ASSERT (*next_event_p++ == CP);
+      return;
+    }
+    case JERRY_PROMISE_CALLBACK_RESOLVE:
+    case JERRY_PROMISE_CALLBACK_REJECT:
+    {
+      TEST_ASSERT (jerry_value_is_promise (object));
+      break;
+    }
+    case JERRY_PROMISE_CALLBACK_BEFORE_REACTION_JOB:
+    case JERRY_PROMISE_CALLBACK_AFTER_REACTION_JOB:
+    {
+      TEST_ASSERT (jerry_value_is_promise (object));
+      TEST_ASSERT (jerry_value_is_undefined (value));
+      break;
+    }
+    case JERRY_PROMISE_CALLBACK_ASYNC_AWAIT:
+    {
+      TEST_ASSERT (jerry_value_is_object (object));
+      TEST_ASSERT (jerry_value_is_promise (value));
+      break;
+    }
+    default:
+    {
+      TEST_ASSERT (notification_type == JERRY_PROMISE_CALLBACK_ASYNC_BEFORE_RESOLVE
+                   || notification_type == JERRY_PROMISE_CALLBACK_ASYNC_BEFORE_REJECT
+                   || notification_type == JERRY_PROMISE_CALLBACK_ASYNC_AFTER_RESOLVE
+                   || notification_type == JERRY_PROMISE_CALLBACK_ASYNC_AFTER_REJECT);
+      TEST_ASSERT (jerry_value_is_object (object));
+      break;
+    }
+  }
+
+  TEST_ASSERT (*next_event_p++ == (uint8_t) notification_type);
+} /* promise_callback */
+
+static void
+run_eval (const uint8_t *event_list_p, /**< event list */
+          const char *source_p) /**< source code */
+{
+  next_event_p = event_list_p;
+
+  jerry_value_t result = jerry_eval ((const jerry_char_t *) source_p, strlen (source_p), 0);
+
+  TEST_ASSERT (!jerry_value_is_error (result));
+  jerry_release_value (result);
+
+  result = jerry_run_all_enqueued_jobs ();
+  TEST_ASSERT (!jerry_value_is_error (result));
+  jerry_release_value (result);
+
+  TEST_ASSERT (*next_event_p == UINT8_MAX);
+} /* run_eval */
+
+int
+main (void)
+{
+  TEST_INIT ();
+
+  if (!jerry_is_feature_enabled (JERRY_FEATURE_PROMISE))
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Promise is disabled!\n");
+    return 0;
+  }
+
+  jerry_init (JERRY_INIT_EMPTY);
+
+  jerry_promise_set_callback (promise_callback, (void *) &user);
+
+  /* Next test. */
```

Comment:
This does not provide extra information. We can keep it, however describing what are we testing would be way better.

Reply:
Maybe I can give a bit more info, but it is a separator, makes it easier to find the star of a test.